ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -274,6 +270,23 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
     isStackedBar,
   ]);
 
+  // Update adjacent charts so tooltip is unpinned when current chart is clicked.
+  useEffect(() => {
+    // Only allow pinning one tooltip at a time, subsequent tooltip click unpins previous.
+    // Multiple tooltips can only be pinned if Ctrl or Cmd key is pressed while clicking.
+    if (tooltipPinnedCoords !== null && lastTooltipPinnedCoords !== null) {
+      if (!isEqual(lastTooltipPinnedCoords, tooltipPinnedCoords)) {
+        setTooltipPinnedCoords(null);
+      }
+      const isCrosshairPinned = checkCrosshairPinnedStatus(seriesMapping);
+      if (tooltipPinnedCoords !== null && isCrosshairPinned) {
+        seriesMapping.pop();
```

Comment:
Is this directly modifying an array-based variable that was passed in as a prop or am I missing some other stuff going on here? If I recall correctly, it's risky to directly mutate props in react codebases.

Reply:
Yeah was rushing a bit to get it usable, just refactored the approach, although need to test performance again with [this change](https://github.com/perses/perses/blob/27c7055cde60220d56de379aa18dcf4b9d785870/ui/components/src/TimeChart/TimeChart.tsx#L208)